### PR TITLE
[bitnami/mysql] feat: add headless services and fix a helper

### LINKED ISSUE #4347: [bitnami/mysql] Allow setting clusterIP to "None" for headless service
**Which chart**:
bitnami/mysql (version 6.14.11)

**Describe the bug**
As described in StatefulSets docs [here](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations), StatefulSets **require** a headless service for network identity of pods. The helm chart should either default to `clusterIP: "None"` when creating a service of type ClusterIP or allow user to override the value. We are facing random connection reset errors when there are idle connections in the pool.

**To Reproduce**
Deploy chart with default values. 

**Expected behavior**
MySQL should work without connection issues out of the box.

**Version of Helm and Kubernetes**:

```
version.BuildInfo{Version:"v3.4.0", GitCommit:"7090a89efc8a18f3d8178bf47d2462450349a004", GitTreeState:"dirty", GoVersion:"go1.15.3"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:47:43Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
N/A

### LINKED ISSUE #4539: [mysql] unable to create mysql with initdbScripts
**Which chart**:
mysql 8.0.0

**Describe the bug**
unable to create mysql with  initdbScripts, it works if I remove setting for initdbScripts.

The error is:
`Error: template: mysql/templates/primary/statefulset.yaml:283:21: executing "mysql/templates/primary/statefulset.yaml" at <include "mysql.initdbScriptsCM" .>: error calling include: template: mysql/templates/_helpers.tpl:51:34: executing "mysql.initdbScriptsCM" at <include "mysql.master.fullname" .>: error calling include: template: no template "mysql.master.fullname" associated with template "gotpl`

**To Reproduce**
helm install -f MySQLValues.yaml mysql bitnami/mysql
MySQLValues.yaml
```
auth:
  rootPassword: 1qaz2wsx!@#$

primary:
  nodeSelector: 
    kubernetes.io/os: linux
  
initdbScripts:
  createdb.sql: |-
    CREATE DATABASE IF NOT EXISTS test DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_general_ci;
```

**Expected behavior**
Able to create mysql with initdbScripts.

**Version of Helm and Kubernetes**:
Helm 3.4.1
Kubernetes 1.19.3

### PR DESCRIPTION
**Pull request by dani8art**
Signed-off-by: darteaga <darteaga@vmware.com>

**Description of the change**

- Add headless services to primary and secondary statefulset
- update "master" to "primary" in "mysql.initdbScriptsCM" helper

**Benefits**

We can use pods DNS names 

**Possible drawbacks**

N/A

**Applicable issues**

  - fixes #4347
  - fixes #4539

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Review by FraPazGal — APPROVED** [review]
Thanks for the PR! LGTM!

File: bitnami/mysql/templates/secondary/svc-headless.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{- if eq .Values.architecture "replication"}}
```
Comment (marcosbc) [inline bitnami/mysql/templates/secondary/svc-headless.yaml:1]: For consistency:

```suggestion
{{- if eq .Values.architecture "replication" }}
```

**Review by marcosbc — APPROVED** [review]
I cannot see anything wrong with the PR that could be actively causing issues with the chart, and the PR looks good to me. Feel free to merge whenever you consider more appropriate.

File: bitnami/mysql/templates/secondary/svc-headless.yaml (hunk 1 above)
Comment (dani8art) [reply bitnami/mysql/templates/secondary/svc-headless.yaml:1]: yes, in fact, this will fails. Good catch!

**Review by marcosbc — APPROVED** [review]

**Pull request merged**